Makefile

```diff
@@ -651,7 +651,7 @@ fmt:
 vet:
 	$(CONTAINERIZED) $(CALICO_BUILD) \
 	sh -c '$(GIT_CONFIG_SSH) \
-	go vet ./...'
+	go vet -composites=false ./...'
```

Comment:
Why this change? We should just fix unkeyed composites in the code. Does this PR introduce any?

Reply:
Sorry I meant to revisit this, I added it to make things work. I had exported some structs used in our tests but I reverted the `go vet` change and refactored the test structs.

9325aadc84447b18cab320a30181f6a764cd2bd2